projects/demo/src/modules/customization/viewport/examples/2/portal-host.ts

```diff
@@ -0,0 +1,49 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    Inject,
+    INJECTOR,
+    Injector,
+} from '@angular/core';
+import {
+    AbstractTuiPortalHostComponent,
+    AbstractTuiPortalService,
+    TuiDropdownPortalService,
+} from '@taiga-ui/cdk';
+import {tuiAsViewport, TuiRectAccessor} from '@taiga-ui/core';
+
+@Component({
+    selector: `[portalHost]`,
+    template: `
+        <ng-content></ng-content>
+        <ng-container #viewContainer></ng-container>
+    `,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {provide: AbstractTuiPortalService, useExisting: TuiDropdownPortalService},
```

Comment:
Let's use `useClass` instead so it provides a new instance and remove `providers: [TuiDropdownPortalService],` from the example. This way we can drop `TypeScript` from that example completely so there are less tabs.

Reply:
`can drop TypeScript` but we need it for stackblitz